lib/server.js

```diff
@@ -48,6 +48,11 @@ function createServer (options, httpHandler) {
     } else {
       listenOptions.cb = cb
     }
+    if (listenOptions.signal) {
+      const onAborted = () => { this.close() }
+      listenOptions.signal.addEventListener('abort', onAborted)
+      server.once('close', () => listenOptions.signal.removeEventListener('abort', onAborted))
+    }
```

Comment:
like this @metcoder95 ?

```suggestion
    if (listenOptions.signal) {
      if (listenOptions.signal.aborted) {
        this.close()
      } else {
        const onAborted = () => { this.close() }
        listenOptions.signal.addEventListener('abort', onAborted)
        server.once('close', () => listenOptions.signal.removeEventListener('abort', onAborted))
      }
    }
```

Reply:
LGTM 👍 

One thing to have in mind as well is that we need to pass the `{ once: true }` or use the `once` method (if an EventEmitter is passed) to ensure the callback is called just once and removed afterwards

Reply:
Added `{ once: true }`
Signal can be EventEmitter?

Reply:
Not by "default", but if using < v16 and still want to use `AbortController`, there are several polyfill libs that do this for compatibility.

That's why several libs specifies that they accept either the native version, its `EventTarget`, or an `EventEmitter` equivalent